app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -118,4 +122,32 @@ object BitcoindRpcBackendUtil extends BitcoinSLogger {
       }
     }
   }
+
+  def startBitcoindBlockPolling(
+      wallet: Wallet,
+      bitcoind: BitcoindRpcClient,
+      startCount: Int,
+      interval: FiniteDuration = 10.seconds)(implicit
+      system: ActorSystem,
+      ec: ExecutionContext): Cancellable = {
+    var prevCount = startCount
+    system.scheduler.scheduleWithFixedDelay(0.seconds, interval) { () =>
+      {
+        bitcoind.getBlockCount.flatMap { count =>
+          if (prevCount != count) {
```

Comment:
Is this always a safe assumption? Or do we need to individually consider the cases

1. prevCount < count
2. prevcount > count
3. prevCount == count

I'm particularily concerned about 2. Is this possible to hit on testnet?

Reply:
2 is technically possible but extremely unlikely. Can add a case for it though